1873 handle commit tx

prajwolrg opened the pull request:
## Description

<!--
Provide a brief summary of the changes and the motivation behind them.
-->

### Type of Change

<!--
Select the type of change your PR introduces (put an `x` in all that apply):
-->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature/Enhancement (non-breaking change which adds functionality or enhances an existing one)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactor
- [ ] New or updated tests
- [ ] Dependency Update

## Notes to Reviewers

<!--
Anything in particular you want to note that will help reviewers fulfill their role
in reviewing this PR?
-->

## Checklist

<!--
Ensure all the following are checked:
-->

- [ ] I have performed a self-review of my code.
- [ ] I have commented my code where necessary.
- [ ] I have updated the documentation if needed.
- [ ] My changes do not introduce new warnings.
- [ ] I have added (where necessary) tests that prove my changes are effective or that my feature works.
- [ ] New and existing tests pass with my changes.
- [ ] I have [disclosed my use of AI](https://github.com/alpenlabs/alpen/blob/main/CONTRIBUTING.md#ai-assistance-notice) in the body of this PR.

## Related Issues

<!--
Link any related issues (e.g., `closes #123`, `fixes #456`).
-->

barakshani commented on crates/asm/subprotocols/bridge-v1/src/state/fulfillment.rs:
```diff
@@ -0,0 +1,261 @@
+/// fronted the withdrawal transaction for a specific deposit, but has not yet
+/// submitted a commit transaction to claim their payout.
+///
+/// Once the operator submits a commit transaction and the payout is processed,
```
```suggestion
/// Once the operator's commit transaction has been posted onchain and the payout is processed,
```

barakshani commented on crates/asm/subprotocols/bridge-v1/src/state/fulfillment.rs:
```diff
@@ -0,0 +1,261 @@
+///
+/// This table maintains records of withdrawals that have been successfully fulfilled
+/// by operators but for which the operators have not yet submitted commit transactions
+/// to claim their payouts. The table provides efficient insertion, lookup, and removal
```
```suggestion
/// by operators, for which the operators have not yet submitted commit transactions.
/// The table provides efficient insertion, lookup, and removal
```
This is confusing because in most cases a _commit tx_ would have been posted.

barakshani commented on crates/asm/subprotocols/bridge-v1/src/state/fulfillment.rs:
```diff
@@ -0,0 +1,261 @@
+///
+/// - **Add**: When an operator successfully fronts a withdrawal transaction
+/// - **Query**: To check if an operator is eligible to claim a payout for a specific deposit
+/// - **Remove**: When an operator submits a commit transaction and receives their payout
```
```suggestion
/// - **Remove**: When an operator submits a commit transaction
```
commit isn't a payout.

barakshani commented on crates/asm/subprotocols/bridge-v1/src/state/fulfillment.rs:
```diff
@@ -0,0 +1,261 @@
+/// # Fulfillment Workflow
+///
+/// - **Add**: When an operator successfully fronts a withdrawal transaction
+/// - **Query**: To check if an operator is eligible to claim a payout for a specific deposit
```
Query is `contains()`? what do we need this for?

prajwolrg replied:
This doesn't really need to go into docstring.

barakshani commented on crates/asm/subprotocols/bridge-v1/src/state/fulfillment.rs:
```diff
@@ -0,0 +1,261 @@
+
+/// Represents an operator's claim to unlock a deposit UTXO after successful withdrawal processing.
+///
+/// This structure is created when an operator successfully processes a withdrawal by making
```
withdrawals --> withdrawal requests

barakshani commented on crates/asm/subprotocols/bridge-v1/src/state/fulfillment.rs:
```diff
@@ -0,0 +1,261 @@
+/// corresponding locked deposit funds.
+///
+/// The claim contains all necessary information to:
+/// - Link the withdrawal transaction to the original deposit
```
```suggestion
/// - Link the withdrawal fulfillment transaction to the original deposit

barakshani commented on crates/asm/subprotocols/bridge-v1/src/state/fulfillment.rs:
```diff
@@ -0,0 +1,261 @@
+///
+/// The claim contains all necessary information to:
+/// - Link the withdrawal transaction to the original deposit
+/// - Identify which operator performed the withdrawal
```
```suggestion
/// - Identify which operator performed the withdrawal fulfillment

barakshani commented on crates/asm/subprotocols/bridge-v1/src/state/fulfillment.rs:
```diff
@@ -0,0 +1,261 @@
+/// The claim contains all necessary information to:
+/// - Link the withdrawal transaction to the original deposit
+/// - Identify which operator performed the withdrawal
+/// - Enable the Bridge proof to verify the operator's right to withdraw locked funds
```
```suggestion
/// - Enable the bridge proof to prove the operator's right to make a withdrawal from the brige

barakshani commented on crates/asm/subprotocols/bridge-v1/src/state/fulfillment.rs:
```diff
@@ -0,0 +1,261 @@
+/// corresponding deposit UTXOs.
+#[derive(Debug, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
+pub struct OperatorClaimUnlock {
+    /// The transaction idx of the deposit that was assigned.
```
```suggestion
    /// The transaction index of the assigned deposit UTXO.

barakshani commented on crates/asm/subprotocols/bridge-v1/src/subprotocol.rs:
```diff
@@ -42,6 +44,29 @@ impl Subprotocol for BridgeV1Subproto {
+        // Pre-Process each transaction
+        for tx in txs {
+            // Parse transaction to extract structured data, then handle the preprocess transaction
+            // to get the auxilary requests
```
```suggestion
            // to get the auxiliary requests

barakshani commented on crates/asm/subprotocols/bridge-v1/src/subprotocol.rs:
```diff
@@ -86,15 +111,20 @@ impl Subprotocol for BridgeV1Subproto {
         relayer: &mut impl MsgRelayer,
         _params: &Self::Params,
     ) {
+        // Compute the expected N/N locking script once to avoid multiple calculations
```
```suggestion
        // Compute the expected N/N multisig address once to avoid multiple calculations

barakshani commented on crates/asm/subprotocols/bridge-v1/src/subprotocol.rs:
```diff
@@ -86,15 +111,20 @@ impl Subprotocol for BridgeV1Subproto {
         _params: &Self::Params,
     ) {
+        // Compute the expected N/N locking script once to avoid multiple calculations
+        let nn_script = ScriptBuf::new_p2tr(SECP256K1, state.operators().agg_xonly(), None);
```
`nn_address`?

barakshani commented on crates/asm/txs/bridge-v1/src/commit/parse.rs:
```diff
@@ -0,0 +1,244 @@
+pub struct CommitInfo {
+    /// The index of the deposit that the operator is committing to.
+    /// This must be validated against the operator's assigned deposits in the state's assignments
+    /// table to ensure the operator is authorized to commit to this specific deposit.
```
```suggestion
    /// table to ensure the operator is authorized to withdraw this specific deposit.

prajwolrg commented on crates/asm/subprotocols/bridge-v1/src/state/fulfillment.rs:
```diff
@@ -0,0 +1,261 @@
+///
+/// This represents a completed withdrawal where an operator has successfully
+/// fronted the withdrawal transaction for a specific deposit, but has not yet
+/// submitted a commit transaction to claim their payout.
```
I meant Completed fulfillment

prajwolrg commented:
I've tried addressing the comments since some of the design around validating N/N covenant will be used for other bridge transactions as well. I agree with most of the comments, I'm a bit hesitant to call it N/N `address` because it is not `bitcoin::Address`. We can come to an agreement on that. After that I can close this PR, since we won't handle commit transaction in the first release of the bridge.
